### ISSUE TITLE
Container doesn't get cleaned up

### ISSUE THREAD

**Issue by skorokithakis** [dokku/dokku#3794]
I just had a container not get cleaned up. I had previously cleaned up a bunch of containers with non-dokku names (they had Docker's random_words names), but this container was properly named "myapp-web-1.984197294" and had been around for a month.

I have PMed you the `docker inspect` output.

**Comment by josegonzalez** [comment]
We need to cleanup deployment task containers, and also any renamed containers during the cleanup command...

**Comment by josegonzalez** [comment]
@skorokithakis can you show the output of `dokku report`?

**Comment by reneweteling** [comment]
Same here: scale is set to 1

```
secure.web.1.1543333779       de385d510240        1.863MiB / 3.765GiB
secure.web.1.1543331718       071fccdf5342        1.867MiB / 3.765GiB
secure.web.1.1543333142       e79623ddb39e        1.867MiB / 3.765GiB
secure.web.1                  918b342e6eca        1.871MiB / 3.765GiB
secure.web.1.1543333443       47e9793f8205        1.875MiB / 3.765GiB
```

**Comment by josegonzalez** [comment]
@reneweteling if you are talking about the `secure` containers, they might be due to being created on a much older version of Dokku that did not keep track of older containers. You can manually delete them now, and future temp containers _should_ be removed.

**Comment by reneweteling** [comment]
Hi @josegonzalez,

I was indeed talking about weteling.web and secure.web, i updated to the latest version `0.19.11`. Ive stopped the lingurring containers using `docker stop <container_id>` then i did the following:

## docker ps | grep weteling.web
```
CONTAINER ID        IMAGE                       COMMAND                  CREATED             STATUS                 PORTS               NAMES
6e5df162288e        dokku/weteling:latest      "bundle exec puma -C…"   34 minutes ago      Up 20 seconds       5000/tcp            weteling.web.1
```
## reboot
`shutdown -r now`
## docker ps
```
CONTAINER ID        IMAGE                       COMMAND                  CREATED             STATUS                 PORTS               NAMES
6e5df162288e        dokku/weteling:latest      "bundle exec puma -C…"   34 minutes ago      Up 20 seconds       5000/tcp            weteling.web.1
2e545d59d727        c578bf37b577               "/start web"             23 months ago       Up 20 seconds                           weteling.web.1.1519143748
```

Same goes for the secure app, only than ive got 4 lingurring
```
CONTAINER ID        IMAGE                       COMMAND                  CREATED             STATUS                 PORTS               NAMES
918b342e6eca        dokku/secure:latest        "/bin/sh -c 'nginx -…"   2 hours ago         Up About a minute   5000/tcp            secure.web.1
de385d510240        85222b690f35               "/bin/sh -c 'nginx -…"   13 months ago       Up About a minute   5000/tcp            secure.web.1.1543333779
47e9793f8205        85222b690f35               "/bin/sh -c 'nginx -…"   13 months ago       Up About a minute   5000/tcp            secure.web.1.1543333443
e79623ddb39e        85222b690f35               "/bin/sh -c 'nginx -…"   13 months ago       Up About a minute   5000/tcp            secure.web.1.1543333142
071fccdf5342        85222b690f35               "/bin/sh -c 'nginx -…"   18 months ago       Up About a minute   5000/tcp            secure.web.1.1543331718
```

Any suggestions are welcome :)

**Comment by josegonzalez** [comment]
You need to `rm` them. Docker will start all stopped containers on boot cause it doesn't keep track of container state (why even implement a daemon....).

**Comment by reneweteling** [comment]
@josegonzalez wow... i feel silly now :P

```
docker stop 47e9793f8205 && docker rm 47e9793f8205
docker stop e79623ddb39e && docker rm e79623ddb39e
docker stop 071fccdf5342 && docker rm 071fccdf5342
docker stop 2e545d59d727 && docker rm 2e545d59d727
```

Currently rebooting

and its fixed! Thanks so much!

**Comment by josegonzalez** [comment]
This should be fixed as of 0.23.0 (we're very aggressive about cleaning up old containers now). If someone sees something similar occur on a version higher than that, please open a new issue.
